config/config_test.go

```diff
@@ -0,0 +1,168 @@
+package config
+
+import (
+	"testing"
+	"github.com/stretchr/testify/assert"
+	"reflect"
+	"gopkg.in/yaml.v2"
+	"github.com/gruntwork-io/boilerplate/errors"
+)
+
+func TestParseBoilerplateConfigEmpty(t *testing.T) {
+	t.Parallel()
+
+	actual, err := ParseBoilerplateConfig([]byte(""))
+	expected := &BoilerplateConfig{}
+
+	assert.Nil(t, err)
+	assert.Equal(t, expected, actual)
+}
+
+func TestParseBoilerplateConfigInvalid(t *testing.T) {
+	t.Parallel()
+
+	_, err := ParseBoilerplateConfig([]byte("not-a-valid-yaml-file"))
+
+	assert.NotNil(t, err)
+
+	unwrapped := errors.Unwrap(err)
+	_, isYamlTypeError := unwrapped.(*yaml.TypeError)
+	assert.True(t, isYamlTypeError, "Expected a YAML type error for an invalid yaml file but got %s", reflect.TypeOf(unwrapped))
```

Comment:
Will this code execute if `err != nil`? Doesn't `assert.NotNil(t, err)` fail the test immediately if `err != nil`?